pepr/robustness/foolbox_wrapper.py

```diff
@@ -0,0 +1,533 @@
+"""PePR wrapper classes for foolbox attack classes."""
+
+import logging
+import numpy
+import numpy as np
+import os
+
+from pepr.attack import Attack
+from pepr import report
+import matplotlib.pyplot as plt
+from pylatex import Command, NoEscape, Tabular, MiniPage
+from pylatex.section import Subsubsection
+from pylatex.utils import bold
+
+import foolbox as fb
+import tensorflow as tf
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+plt.style.use("default")
+# force line grid to be behind bar plots
+plt.rcParams["axes.axisbelow"] = True
+plt.rcParams["axes.grid"] = True
+plt.rcParams["grid.linestyle"] = ":"
+
+
+class BaseAttack(Attack):
+    """
+    Base foolbox attack class implementing the logic for running the attack and
+    generating a report.
+
+    Parameters
+    ----------
+    attack_alias : str
+        Alias for a specific instantiation of the class.
+    attack_pars : dict
+        Dictionary containing all needed attack parameters:
+
+        * epsilons (list): List of one or more Epsilons for the attack.
+
+    data : numpy.ndarray
+        Dataset with all input images used to attack the target models.
+    labels : numpy.ndarray
+        Array of all labels used to attack the target models.
+    data_conf : dict
+        Dictionary describing for every target model which record-indices should be used
+        for the attack.
+
+        * attack_indices_per_target (numpy.ndarray): Array of indices to
+          attack per target model.
+
+    target_models : iterable
+        List of target models which should be tested.
+    foolbox_attack : foolbox.attack.Attack
+        The foolbox attack object which is wrapped in this class.
+    pars_descriptors : dict
+            Dictionary of attack parameters and their description shown in the attack
+            report.
```

Comment:
Can we add an example of a pars_descriptor here? :)

Reply:
Added in 0e3fac3